src/metabase/upload.clj

```diff
@@ -248,7 +248,8 @@
   [{:keys [number-separators] :as _settings}]
   (let [int?          (regex-matcher (int-regex number-separators))
         float-or-int? (regex-matcher (float-or-int-regex number-separators))
-        float?        (regex-matcher (float-regex number-separators))]
+        only-float?   (regex-matcher (float-regex number-separators))
+        float?        #(or (only-float? %) (int? %))]
```

Comment:
hmmm what's the difference between `float?` and `float-or-int?`

Reply:
A `float-or-int?` is something like `3.0` - looks like a float, but technically an integer 😄 

The reason for the distinction is that we can cast these down if inserting into an integer column, but if there is non-zero integral part we want to first alter the column to float so that we don't truncate the value. It was introduced [here](https://github.com/metabase/metabase/pull/40275).